internal/engine/rtengine/cache.go

```diff
@@ -73,17 +76,20 @@ func NewRuleEngineCache(
 	// Populate the cache with rule type engines for the rule types we found.
 	engines := make(cacheType, len(ruleTypes))
 	for _, ruleType := range ruleTypes {
-		ruleEngine, err := cacheRuleEngine(ctx, &ruleType, provider, ingestCache, engines)
+		ruleEngine, err := cacheRuleEngine(ctx, &ruleType, provider, ingestCache, engines, opts...)
 		if err != nil {
 			return nil, err
 		}
 		engines[ruleType.ID] = ruleEngine
 	}
 
-	return &ruleEngineCache{engines: engines}, nil
+	return &ruleEngineCache{engines: engines, opts: opts}, nil
 }
 
-func (r *ruleEngineCache) GetRuleEngine(ctx context.Context, ruleTypeID uuid.UUID) (*RuleTypeEngine, error) {
+func (r *ruleEngineCache) GetRuleEngine(
+	ctx context.Context,
+	ruleTypeID uuid.UUID,
+) (*RuleTypeEngine, error) {
```

Comment:
What's causing this reformat?  It's polluting the diff a bit.

I wouldn't fix things for this PR, but it would be nice to avoid these extra diffs for both blame reasons and ease of PR review.

Reply:
This is due to some earlier experiments that made the signature excede 100 characters, I can revert this.